src/util/handlers/StaticHandler.ts

```diff
@@ -0,0 +1,22 @@
+import { AsyncHandler } from './AsyncHandler';
+
+/**
+ * A handler that always resolves and always returns the stored value.
+ * Will return undefined if no value is stored.
+ *
+ * The generic type extends `any` due to Components.js requirements.
+ */
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+export class StaticHandler<T extends any = void> extends AsyncHandler<any, T> {
+  private readonly value?: T;
+
+  public constructor(value?: T) {
```

Comment:
Why is `undefined` required, and the `?` and `!`?

I'd say it's up to the consumer to define `T` such that it includes `undefined` if they want to.

Reply:
I actually only wanted a handler that takes no input and does nothing, but then extended it to potentially take a value as input because why not. The reason `value` is optional is because otherwise Components.js is going to complain that the variable need to be defined.